File: web/public/avatar/inspect_gltf.js
// inspect_gltf.js
const fs = require('fs');
const p = './avatar.gltf'; // run next to avatar.gltf
if (!fs.existsSync(p)) { console.error('avatar.gltf not found in current folder'); process.exit(1); }
const g = JSON.parse(fs.readFileSync(p,'utf8'));
const anims = g.animations || [];
console.log('ANIMATIONS_COUNT=' + anims.length);
anims.forEach((a,i)=> console.log(`[${i}] name="${a.name||'<unnamed>'}" channels=${(a.channels||[]).length} samplers=${(a.samplers||[]).length}`));
const buffers = g.buffers||[];
buffers.forEach((b,i)=> console.log(`[BUFFER ${i}] uri=${b.uri||'<embedded>'} byteLength=${b.byteLength}`));
const imgs = g.images||[];
imgs.forEach((im,i)=> console.log(`[IMAGE ${i}] uri=${im.uri||'<none>'}`));

File: web/public/avatar/inspect.gltf.py
import json, os
p='./avatar.gltf'
if not os.path.exists(p):
    print("avatar.gltf not found in current folder")
    raise SystemExit(1)
g=json.load(open(p,'r',encoding='utf8'))
anims=g.get('animations',[])
print("ANIMATIONS_COUNT="+str(len(anims)))
for i,a in enumerate(anims):
    print(f"[{i}] name={a.get('name','<unnamed>')} channels={len(a.get('channels',[]))} samplers={len(a.get('samplers',[]))}")
for i,b in enumerate(g.get('buffers',[])):
    print(f"[BUFFER {i}] uri={b.get('uri')} byteLength={b.get('byteLength')}")
for i,im in enumerate(g.get('images',[])):
    print(f"[IMAGE {i}] uri={im.get('uri')}")
print("FILES_IN_FOLDER:")
print('\\n'.join(sorted(os.listdir('.'))))
